src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/Razor/ComponentAccessibilityCodeActionProvider.cs

```diff
@@ -47,7 +47,10 @@ public ComponentAccessibilityCodeActionProvider(ITagHelperFactsService tagHelper
 
         // Find start tag. We allow this code action to work from anywhere in the start tag, which includes
         // embedded C#, so we just have to traverse up the tree to find a start tag if there is one.
-        var startTag = (MarkupStartTagSyntax?)node.FirstAncestorOrSelf<SyntaxNode>(n => n is MarkupStartTagSyntax);
+        // We also check for tag helper start tags here, because an invalid start tag with a valid tag helper
+        // anywhere in it  would otherwise not match. We rely on the IsTagUnknown method below, to ensure we
+        // only offer on actual potential component tags (because it checks for the compiler diagnostic)
+        var startTag = (IStartTagSyntaxNode?)node.FirstAncestorOrSelf<SyntaxNode>(n => n is MarkupStartTagSyntax or MarkupTagHelperStartTagSyntax);
```

Comment:
Is there a reason this isn't just `n is IStartTagSyntaxNode`?

Reply:
Oops, nope, I meant to do that. Thanks!